File: app/src/main/java/com/antelope/goodbrother/business/main/MainActivity.java
package com.antelope.goodbrother.business.main;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.ServiceConnection;
import android.os.Bundle;
import android.os.IBinder;

import com.nirvana.zmkj.base.BaseActivity;

public class MainActivity extends BaseActivity {

    private ServiceConnection serviceConnection;

    private TestService testService;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(new MainViewHolder(this).getRootView());

        Intent intent = new Intent(MainActivity.this, TestService.class);
        startService(intent);

        serviceConnection = new ServiceConnection() {
            @Override
            public void onServiceConnected(ComponentName name, IBinder service) {

                testService = ((TestService.MyBinder) service).getService();

            }

            @Override
            public void onServiceDisconnected(ComponentName name) {

            }
        };
        bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
    }


    public void getNum() {

        testService.getNum();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unbindService(serviceConnection);
    }
}

File: app/src/main/java/com/antelope/goodbrother/MainActivity.kt
package com.antelope.goodbrother

import android.os.Bundle
import com.nirvana.zmkj.base.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Example of a call to a native method
        sample_text.text = "aaaa"
    }
}
